src/services/security/modules/RedefineModule/index.ts

```diff
@@ -0,0 +1,154 @@
+import { generateTypedData } from '@safe-global/safe-core-sdk-utils'
+import { type SecurityResponse, type SecurityModule, SecuritySeverity } from '../types'
+
+export const enum REDEFINE_ERROR_CODES {
+  ANALYSIS_IN_PROGRESS = 1000,
+  SIMULATION_FAILED = 1001,
+  INPUT_VALIDATION = 2000,
+  BAD_REQUEST = 3000,
+}
+
+const redefineSeverityMap: Record<RedefineSeverity['label'], SecuritySeverity> = {
+  CRITICAL: SecuritySeverity.CRITICAL,
+  HIGH: SecuritySeverity.HIGH,
+  MEDIUM: SecuritySeverity.MEDIUM,
+  LOW: SecuritySeverity.LOW,
+  NO_ISSUES: SecuritySeverity.NONE,
+}
+
+export type RedefineModuleRequest = {
+  chainId: number
+  safeAddress: string
+  walletAddress: string
+  safeTransaction: SafeTransaction
+  threshold: number
+}
+
+export type RedefineModuleResponse = {
+  issues?: Array<
+    Omit<NonNullable<RedefineResponse['data']>['insights']['issues'][number], 'severity'> & {
+      severity: SecuritySeverity
+    }
+  >
+  balanceChange?: NonNullable<RedefineResponse['data']>['balanceChange']
+  simulation?: NonNullable<RedefineResponse['data']>['simulation']
+  errors: RedefineResponse['errors']
+}
+
+type RedefinePayload = {
+  chainId: number
+  domain?: string
+  payload: {
+    method: 'eth_signTypedData_v4'
+    params: [string, string]
+  }
+}
+
+type RedefineSeverity = {
+  code: number
+  label: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW' | 'NO_ISSUES'
+}
+
+type RedefineBalanceChange =
+  | {
+      address: string
+      amount: {
+        value: string
+        normalizedValue: string
+      }
+      type: 'ERC20'
+      symbol: string
+      decimals: number
+      name: string
+    }
+  | {
+      amount: {
+        value: string
+        normalizedValue: string
+      }
+      type: 'NATIVE'
+      symbol: string
+      decimals: number
+      name: string
+    }
+  | { type: 'ERC721'; address: string; tokenId: string; name?: string; symbol?: string }
+
+export type RedefineResponse = {
+  data?: {
+    balanceChange?: {
+      in: RedefineBalanceChange[]
+      out: RedefineBalanceChange[]
+    }
+    insights: {
+      issues: {
+        description: {
+          short: string
+          long: string
+        }
+        severity: RedefineSeverity
+        category: string
+      }[]
+      verdict: RedefineSeverity
+    }
+    simulation: {
+      uuid: string
+      time: string
+      block: string
+    }
+  }
+  errors: {
+    code: number
+    message: string
+    extendedInfo?: Record<string, unknown>
+  }[]
+}
+
+export class RedefineModule implements SecurityModule<RedefineModuleRequest, RedefineModuleResponse> {
+  async scanTransaction(request: RedefineModuleRequest): Promise<SecurityResponse<RedefineModuleResponse>> {
+    const { chainId, safeAddress } = request
+
+    const txTypedData = generateTypedData({
+      safeAddress,
+      safeVersion: '1.3.0', // TODO: pass to module, taking into account that lower Safe versions don't have chainId in payload
```

Comment:
Should this be tackled here?

Reply:
Good question. I remember there being an issue with passing a different safeVersion and it worked even with 1.1.1 safes so we decided to postpone it but this is a good time to revisit this.

Reply:
I guess the `TODO` is a good marker for us to revisit this. We can look at it when porting it to the new flow.

I'll resolve this for the time being